src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexReplacement.cs

```diff
@@ -38,7 +39,8 @@ public RegexReplacement(string rep, RegexNode concat, Hashtable _caps)
 
             Span<char> vsbStack = stackalloc char[256];
             var vsb = new ValueStringBuilder(vsbStack);
-            var strings = new List<string>();
+            FourStackStrings stackStrings = default;
+            var strings = new ValueListBuilder<string>(MemoryMarshal.CreateSpan(ref stackStrings.Item1!, 4));
```

Comment:
`ref stackStrings.Item1!`

This is kind of surprising - specifically the `stackStrings.Item1!` part. `Item1` is `null` because `stackStrings` is set to `default` on the line before.

Reply:
ValueListBuilder just needs scratch stack space; it's not actually looking at any of the values passed in.  Suppressing the null checking seemed the simplest answer.

Reply:
Is there an underlying compiler/language issue here? Taking a `ref` of a null reference should be OK, right?

```C++
   void *p = nullptr;
   void **pp = &p;
```

Reply:
It's not that taking a ref to it is a problem; it's that it's essentially then got a `ref string?` and is passing that to something typed as `ref string`; if the thing expecting a non-nullable string then dereferenced it and got back null, it'd be violating the non-nullability.